src/main/java/com/zendesk/maxwell/schema/columndef/DateFormatter.java

```diff
@@ -33,17 +34,17 @@ public static Timestamp extractTimestamp(Object value) {
 			return new Timestamp(time);
 		} else
 			throw new RuntimeException("couldn't extract date/time out of " + value);
-
 	}
 
-	private static Long MIN_DATE = Timestamp.valueOf("1000-01-01 00:00:00").getTime();
-	private static String extractAndFormat(SimpleDateFormat formatter, Object value) {
-		synchronized(formatter) {
-			Timestamp t = extractTimestamp(value);
-			if ( t.getTime() < MIN_DATE )
-				return null;
-			else
-				return formatter.format(t);
+	private static Timestamp MIN_DATE = Timestamp.valueOf("1000-01-01 00:00:00");
+
+	private static String format(SimpleDateFormat formatter, Timestamp ts) {
+		if ( ts.before(MIN_DATE) ) {
+			return null;
```

Comment:
An exception would stop maxwell from processing, which is probably not what we want (the data won't get any better; we have to do something and continue).

From 089d76bf19e77dee67097a7b64dd2a8381e3b80c (when this check was added) it sounds like such dates are invalid, so `null` ought to be fine:

> The only non-refactoring change here deals with how we deal with
"invalid" dates; those before '1000-01-01'.  Mysql's documentation seems
to indicate that dates like '0000-00-00' should be considered invalid,
although it accepts the damn things anyway.  Maxwell has had highly
differing policies on the data in the past; invalid dates coming from
the bootstrap code paths were null, dates from 5.5 were reproduced
faithfully, and dates from 5.6 were off by a few years.
binlog-connector makes 'em null, and that's good enough for me.